packages/xstate-vue/src/useMachine.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  ref,
+  Ref,
+  watch,
+  onBeforeMount,
+  onBeforeUnmount
+} from '@vue/composition-api';
+import {
+  interpret,
+  EventObject,
+  StateMachine,
+  State,
+  Interpreter,
+  InterpreterOptions,
+  MachineOptions,
+  StateConfig
+} from 'xstate';
+
+interface UseMachineOptions<TContext, TEvent extends EventObject> {
+  /**
+   * If provided, will be merged with machine's `context`.
+   */
+  context?: Partial<TContext>;
+  /**
+   * The state to rehydrate the machine to. The machine will
+   * start at this state instead of its `initialState`.
+   */
+  state?: StateConfig<TContext, TEvent>;
+}
+
+const defaultOptions = {
+  immediate: false
+};
+
+export function useMachine<TContext, TEvent extends EventObject>(
+  machine: StateMachine<TContext, any, TEvent>,
+  options: Partial<InterpreterOptions> &
+    Partial<UseMachineOptions<TContext, TEvent>> &
+    Partial<MachineOptions<TContext, TEvent>> = defaultOptions
+): {
+  current: Ref<State<TContext, TEvent>>;
+  send: Interpreter<TContext, any, TEvent>['send'];
+  service: Ref<Interpreter<TContext, any, TEvent>>;
+} {
+  const {
+    context,
+    guards,
+    actions,
+    activities,
+    services,
+    delays,
+    state: rehydratedState,
+    ...interpreterOptions
+  } = options;
+
+  const machineConfig = {
+    context,
+    guards,
+    actions,
+    activities,
+    services,
+    delays
+  };
+
+  const machineRef = ref<StateMachine<TContext, any, TEvent>>(
+    machine.withConfig(machineConfig, {
+      ...machine.context,
+      ...context
+    } as TContext)
+  );
+
+  const service = ref<Interpreter<TContext, any, TEvent>>(
+    interpret(machineRef.value, interpreterOptions).onTransition(state => {
+      if (state.changed) {
+        current.value = state;
+      }
+    })
+  );
+
+  const initialState = rehydratedState
+    ? State.create(rehydratedState)
+    : service.value.initialState;
+
+  const current = ref<State<TContext, TEvent>>(initialState);
+
+  // Make sure actions and services are kept updated when they change.
+  watch(() => {
```

Comment:
are those needed? are `actions` reactive in here?

Reply:
since actions are not wrapped with ref() or reactive() they are not reactive.

Reply:
they don't need to be watched here then, right?

Reply:
If wewant to replicate the React Hooks behaviour I think they should be watched, but then we'd have to think about if the options should be provided as refs or converted to refs on the initialization.

Reply:
Do we need to replicate it? Those lines in react hooks version are only meant to fix the stale closure problem, can Vue version suffer from this as well?